src/pal/src/include/pal/sharedmemory.h

```diff
@@ -0,0 +1,247 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _PAL_SHARED_MEMORY_H_
+#define _PAL_SHARED_MEMORY_H_
+
+#include "corunix.hpp"
+
+#ifndef static_assert_no_msg
+#define static_assert_no_msg( cond ) static_assert( cond, #cond )
+#endif // !static_assert_no_msg
+
+#ifndef _countof
+#define _countof(a) (sizeof(a) / sizeof(a[0]))
+#endif // !_countof
+
+#ifndef PAGE_SIZE
+#define PAGE_SIZE (0x1000)
+#endif // !PAGE_SIZE
+
+// - Global shared memory files go in:
+//     /tmp/coreclr/shm/global/<fileName>
+// - Session-scoped shared memory files go in:
+//     /tmp/coreclr/shm/session<sessionId>/<fileName>
+// - Lock files associated with global shared memory files go in:
+//     /tmp/coreclr/lockfiles/global/<fileName>
+// - Lock files associated with session-scoped shared memory files go in:
+//     /tmp/coreclr/lockfiles/session<sessionId>/<fileName>
+
+#define SHARED_MEMORY_MAX_FILE_NAME_CHAR_COUNT (_MAX_FNAME - 1)
+#define SHARED_MEMORY_MAX_NAME_CHAR_COUNT (_countof("Global\\") - 1 + SHARED_MEMORY_MAX_FILE_NAME_CHAR_COUNT)
+
+#define SHARED_MEMORY_TEMP_DIRECTORY_PATH "/tmp"
+#define SHARED_MEMORY_RUNTIME_TEMP_DIRECTORY_PATH "/tmp/coreclr"
+
+#define SHARED_MEMORY_SHARED_MEMORY_DIRECTORY_PATH "/tmp/coreclr/shm"
+#define SHARED_MEMORY_LOCK_FILES_DIRECTORY_PATH "/tmp/coreclr/lockfiles"
```

Comment:
This is probably a more general question than this PR, but does coreclr already generate temp files?  We have some in corefx, and I'm wondering if it's important that we align on the locations for these.  corefx uses "/tmp/dotnet/corefx/featurename/...", though the "tmp" isn't hardcoded and instead come from Path.GetTempPath(), which will use the environment variable TMPDIR for the temp path and then fall back to "/tmp/" if that env var wasn't set.